### PR TITLE
Replace MDI icons due to breaking changes in Nerd Fonts

### LINKED ISSUE #72: Need to change the Nerd Font icons that starts with the `nf-mdi` prefix
## Description

The code points have been updated as the material icons are in version 3.0. Because of this, icons such as `nf-mdi-git`, which are currently used as default, are not displayed normally. All default `nf-mdi-*` icons should be replaced.

## Reproduction

```bash
echo "\ufc1d \uf5e1 \uf7a1 \uf7e8 \uf81f \ufc0c \ufcb5 \uf668 \ufb67 \ufa66 \ufd42"
```

## Environment

 - Device: macOS
 - OS: Ventura 13.3.1 (a)
 - Font: JetBrainsMonoNL NF Regular
 - Version: Latest

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by romantech**
## Fixes

Fixes #72  by @romantech

## Description

- Change all Nerd Font icons that start with the `nf-mdi` prefix
- Change icon name comments for the changed icons

Icon Name | From | To
-- | -- | --
audio | \ufc1d | \uf1c7
broom | \uf5e1 | \ue615
git | \uf7a1 | \uf47f
python | \uf81f | \ue73c
runner | \ufc0c | \ueb7b
shell | \ufcb5 | \uf489
source | \uf668 | \uf1c9
test | \ufb67 | \uea79
video | \ufa66 | \uf1c8
audio | \ufc1d | \uf1c7
vuejs | \ufd42 | \ue6a0

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/pls/data/nerd_icons.yml [hunk 1]
```diff
@@ -5,31 +5,31 @@ nerd_icons:
   pls:        "\uf444"  # nf-oct-primitive_dot
 
   apple:      "\ue711"  # nf-dev-apple
-  audio:      "\ufc1d"  # nf-mdi-speaker_wireless
-  broom:      "\uf5e1"  # nf-mdi-broom
+  audio:      "\uf1c7"  # nf-fa-file_audio_o
+  broom:      "\ue615"  # nf-seti-config
```
Comment (dhruvkb) [inline src/pls/data/nerd_icons.yml:9]: There's no other broom, but stars convey the idea of "sparkling clean".
```suggestion
  broom:      "\ue370"  # nf-weather-stars
```

File: src/pls/data/nerd_icons.yml [hunk 2]
```diff
@@ -5,31 +5,31 @@ nerd_icons:
   pls:        "\uf444"  # nf-oct-primitive_dot
 
   apple:      "\ue711"  # nf-dev-apple
-  audio:      "\ufc1d"  # nf-mdi-speaker_wireless
-  broom:      "\uf5e1"  # nf-mdi-broom
+  audio:      "\uf1c7"  # nf-fa-file_audio_o
```
Comment (dhruvkb) [inline src/pls/data/nerd_icons.yml:8]: I find Seti's audio closer to the previous speaker icon.
```suggestion
  audio:      "\ue638"  # nf-seti-audio
```

File: src/pls/data/nerd_icons.yml (hunk 1 above)
Comment (dhruvkb) [inline src/pls/data/nerd_icons.yml:9]: There's no other broom, but stars convey the idea of "sparkling clean".
```suggestion
  broom:      "\ue370"  # nf-weather-stars
```

File: src/pls/data/nerd_icons.yml [hunk 3]
```diff
@@ -5,31 +5,31 @@ nerd_icons:
...
   config:     "\ue615"  # nf-seti-config
   css:        "\ue749"  # nf-dev-css3
   docker:     "\uf308"  # nf-linux-docker
   folder:     "\uf07b"  # nf-fa-folder
-  git:        "\uf7a1"  # nf-mdi-git
+  git:        "\uf47f"  # nf-oct-git_compare
   github:     "\uf408"  # nf-oct-mark_github
   html:       "\ue736"  # nf-dev-html5
-  image:      "\uf7e8"  # nf-mdi-image
+  image:      "\uf1c5"  # nf-fa-file_image_o
```
Comment (dhruvkb) [inline src/pls/data/nerd_icons.yml:17]: Only removing the page flap from the top-right corner.
```suggestion
  image:      "\uf03e"  # nf-fa-image
```

File: src/pls/data/nerd_icons.yml [hunk 4]
```diff
@@ -5,31 +5,31 @@ nerd_icons:
...
   javascript: "\ue60c"  # nf-seti-javascript
   json:       "\ue60b"  # nf-seti-json
   license:    "\uf495"  # nf-oct-law
   lock:       "\uf456"  # nf-oct-lock
   markdown:   "\uf48a"  # nf-oct-markdown
   package:    "\uf487"  # nf-oct-package
   env:        "\ue22f"  # nf-fae-plant
-  python:     "\uf81f"  # nf-mdi-language_python
+  python:     "\ue73c"  # nf-dev-python
```
Comment (dhruvkb) [inline src/pls/data/nerd_icons.yml:25]: Seti's Python has bigger spaces between shapes, which helps differentiate the two pythons at the small font size.
```suggestion
  python:     "\ue606"  # nf-seti-python
```

File: src/pls/data/nerd_icons.yml [hunk 5]
```diff
@@ -5,31 +5,31 @@ nerd_icons:
...
   ruby:       "\ue791"  # nf-dev-ruby_rough
-  runner:     "\ufc0c"  # nf-mdi-run
+  runner:     "\ueb7b"  # nf-cod-vm_running
   sass:       "\ue603"  # nf-seti-sass
-  shell:      "\ufcb5"  # nf-mdi-console_line
-  source:     "\uf668"  # nf-mdi-code_braces
-  test:       "\ufb67"  # nf-mdi-test_tube
+  shell:      "\uf489"  # nf-oct-terminal
+  source:     "\uf1c9"  # nf-fa-file_code_o
```
Comment (dhruvkb) [inline src/pls/data/nerd_icons.yml:30]: Seti's JSON braces are closer to the previous braces icon.
```suggestion
  source:     "\ue60b"  # nf-seti-json
```

File: src/pls/data/nerd_icons.yml [hunk 6]
```diff
@@ -5,31 +5,31 @@ nerd_icons:
...
+  test:       "\uea79"  # nf-cod-beaker
   text:       "\ue612"  # nf-seti-text
   typescript: "\ue628"  # nf-seti-typescript
-  video:      "\ufa66"  # nf-mdi-video
-  vuejs:      "\ufd42"  # nf-mdi-vuejs
+  video:      "\uf1c8"  # nf-fa-file_movie_o
```
Comment (dhruvkb) [inline src/pls/data/nerd_icons.yml:34]: FA's video camera icon is closer to the previous one.
```suggestion
  video:      "\uf03d"  # nf-fa-video_camera
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Thanks for reporting, and fixing the issue @romantech. I appreciate you painstakingly finding replacements for all the MDI icons.

I have some recommended changes to the default set and I hope you won't mind. You can always override the defaults to your preferred icons via the `.pls.yml` config.

**Comment by romantech** [comment]
> Thanks for reporting, and fixing the issue @romantech. I appreciate you painstakingly finding replacements for all the MDI icons.
> 
> I have some recommended changes to the default set and I hope you won't mind. You can always override the defaults to your preferred icons via the `.pls.yml` config.

Thank you for the suggestion. Since I arbitrarily specified the replacement icon, an icon that is closer to the default value you suggested would be more appropriate. Thank you.

**Pull request merged**